### PR TITLE
Add aditional jobs to run non-root implementation

### PR DESCRIPTION
**Pull request by xpivarc**
Add additional jobs to run non-root implementation

/hold
Needs https://github.com/kubevirt/kubevirt/pull/5540 to be merged.

### CODE REVIEW

**Comment by xpivarc** [comment]
/hold cancel
/cc @dhiller @stu-gott 

Kubevirt PR was merged

**Review by dhiller — APPROVED** [review]
Looks good to me. IIUC the lanes are triggered only manually for now, which will minimize additional burden to our CI.

@fgimenez FYI

**Comment by dhiller** [comment]
BTW the PR for kubevirt has already been merged.

**Comment by xpivarc** [comment]
> Looks good to me. IIUC the lanes are triggered only manually for now, which will minimize additional burden to our CI.
> 
> @fgimenez FYI

I want them to run always after the nonroot PR gets merged.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (fgimenez) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:609]:
```suggestion
        - "automation/test.sh"
```

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (fgimenez) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:788]: Could you please move the env vars here to the `env` element and leave here just `automation/test.sh`?

**Review by fgimenez — COMMENTED** [review]
Dropped a couple of comments, I don't think we should enable these jobs to execute on each PR, currently we have 10 bare metals for executing this kind of jobs that are not enough for executing the current load (for instance today we had a queue of 45 jobs for more than 3 hours). We are running now 14 e2e jobs per code push per PR, adding 3 more jobs means increasing the load in  ~21% without increasing the capacity. 

I would wait until we have more capacity for enabling these jobs, in the mean time we could execute them on periodics, wdyt?

**Comment by fgimenez** [comment]
Additionally, we are going to remove the provider lanes (1.19, 1.18, etc) and instead aim to have more focused, per-sig lanes. Besides being more focused and stable, these per-sig lanes take much shorter time to execute, less than 2 hours on average. If we enable the jobs in this PR to run on each code push they would be the longest builds, and would block all the rest.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (xpivarc) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: done

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (xpivarc) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: done

**Comment by xpivarc** [comment]
> Additionally, we are going to remove the provider lanes (1.19, 1.18, etc) and instead aim to have more focused, per-sig lanes. We have observed that there are some tests that only fail in the full suite and not in the sig executions, maybe the state is not properly cleaned or there are some leaks. In any case, introducing lanes that run the whole suite would make the complete build more flaky and the PRs harder to merge.
> 
> Besides being more stable, these per-sig lanes take much shorter time to execute, less than 2 hours on average. If we enable the jobs in this PR to run on each code push they would be the longest builds, and would block all the rest.

Here`s my proposal. Let's merge this now without runalways as suggested. This will enable some testing and we can start discussions on how to test this in long term. I will bring this to the community meeting.

**Comment by fgimenez** [comment]
/lgtm
/approved

> Here`s my proposal. Let's merge this now without runalways as suggested. This will enable some testing and we can start discussions on how to test this in long term. I will bring this to the community meeting.

Ok, sounds good to me.

**Comment by fgimenez** [comment]
@xpivarc and me talked about running the tests with the non-root feature gate enabled on a daily periodic and making the release depend on this lane passing, merging the jobs in this PR will enable testing for now.

**Pull request merged**
